### PR TITLE
chore: scaffold monorepo

### PR DESCRIPTION
**Pull request by sripwoud**
Towards replacing the template rendering engine used for the web interface by a JS/TS application.  
In the internal PSE #dogfood, it seems we were all in favor on using a monorepo to manage the freedit rs backend and the to be developed JS client.  

I picked a few tools that I am setting up in this PR:
- [`bun`](https://bun.sh/docs/installation) as JS runtime and node package manager
- [`just`](https://just.systems) as tasks runner
- [`vite`](https://vite.dev/): JS/TS front end build tool
- [tanstack router](https://tanstack.com/router/latest/docs/framework/react/overview): client side router
  Right now I am just "dangerously setting" the inner html in one route of client app from what the server gives. But of course we should not keep doing that way, this is just for demo/test purposes atm. Indeed atm the server only serves html documents (the rinja template rendering engine). Eventually we want to have a server that gives us JSON and fetch that client side, and let react build the html docs (see #5 and #6).
  
  image

## Test plan
- Spin up the server and client: eg with `just dev`
- you should not see any errors or warnings
- go to [localhost:5173](http://localhost:5173) . Click on the inn link. You should see the same content  as on [localhost:3001](http://localhost:3001) (but displayed very ugly)

### CODE REVIEW

**Review by adria0 — DISMISSED** [review]
Not checked JS stuff because I do not have expertice on this, but overall, LGTM.

File: config.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+db = "freedit.db"
+addr = "127.0.0.1:3001"
+avatars_path = "static/imgs/avatars"
+inn_icons_path = "static/imgs/inn_icons"
+upload_path = "static/imgs/upload"
+tantivy_path = "tantivy"
+proxy = ""
```
Comment (jacque006) [inline config.toml:7]: This file should be in the `apps/server` dir, unless it is now needed by the client?

**Review by jacque006 — DISMISSED** [review]
JS setup looks good to me :+1: 

I assume based on these changes that we will no longer be pulling from freedit upstream. Still might be doable to as custom patches if they make smaller updates. As an example of something we might want: https://github.com/freedit-org/freedit/pull/306

File: README.md [hunk 2]
```diff
@@ -1,70 +1,10 @@
-# freedit
+# PSE Freedit
```
[the remark this answers is not in the document]
Comment (sripwoud) [reply README.md:2]: captured in https://github.com/pseXperiments/freedit/issues/11

File: config.toml (hunk 1 above)
Comment (sripwoud) [reply config.toml]: that file is even supposed to be .gitignored.  
I hadn't noticed all the default dirs and config stuff that get created by `config.rs`. So even though I moved the code in `apps/server`, all that config stuff (`config.toml`, `tantivy/`, `freedit.db` etc...) was still created in the monorepo root dir.
I refactored it in the last commit 
https://github.com/pseXperiments/freedit/pull/4/commits/bbf812cf931fd0b5ac32d94252e477c8c33f9a51#diff-fa36ea6251296f1ff94b66f84cbab988e9ed478733e2a611c3d458ea084ca940

**Review by jacque006 — APPROVED** [review]

**Comment by jacque006** [comment]
We'll need to port this to https://github.com/privacy-scaling-explorations/pse-forums , can likely just push `main`  there. Will also include db snapshot logic from https://github.com/pseXperiments/freedit/pull/1 & https://github.com/pseXperiments/freedit/pull/2

**Comment by sripwoud** [comment]
@jacque006 yes i wanted to merge it here first as you had started to review it here.
I ll push to main on pse-forums

**Pull request merged**